libraries/Adapters/Microsoft.Bot.Builder.Adapters.WeChat/MessageCryptography.cs

```diff
@@ -0,0 +1,168 @@
+            {
+                throw new UnauthorizedAccessException("Signature validation failed.");
+            }
+
+            return AesDecrypt(encryptMessage, _encodingAesKey, _appId);
+        }
+
+        /// <summary>
+        /// Decrypt the message.
+        /// </summary>
+        /// <param name="encryptString">Encrypted string.</param>
+        /// <param name="encodingAesKey">Encoding AES key for descrypt message.</param>
+        /// <param name="appId">The WeChat app id.</param>
+        /// <returns>Decrypted string.</returns>
+        public string AesDecrypt(string encryptString, string encodingAesKey, string appId)
+        {
+            try
+            {
+                var key = Convert.FromBase64String(encodingAesKey + "=");
+                var iv = new byte[16];
+                Array.Copy(key, iv, 16);
+                var btmpMsg = AesDecrypt(encryptString, iv, key);
+
+                var len = BitConverter.ToInt32(btmpMsg, 16);
+                len = IPAddress.NetworkToHostOrder(len);
+
+                var messageBytes = new byte[len];
+                var appIdBytes = new byte[btmpMsg.Length - 20 - len];
+                Array.Copy(btmpMsg, 20, messageBytes, 0, len);
+                Array.Copy(btmpMsg, 20 + len, appIdBytes, 0, btmpMsg.Length - 20 - len);
+                var oriMsg = Encoding.UTF8.GetString(messageBytes);
+                if (appId != Encoding.UTF8.GetString(appIdBytes))
+                {
+                    throw new ArgumentException("Failed to validate appId.", nameof(appId));
+                }
+
+                return oriMsg;
+            }
+            catch (FormatException e)
+            {
+                throw new FormatException("Failed to decode base64 string.", e);
+            }
+        }
+
+        /// <summary>
+        /// Decode the decrypted byte array.
+        /// </summary>
+        /// <param name="decrypted">Decrypted byte array.</param>
+        /// <returns>Decoded byte array.</returns>
+        private static byte[] Decode(byte[] decrypted)
+        {
+            var pad = decrypted[decrypted.Length - 1];
+            if (pad < 1 || pad > 32)
+            {
+                pad = 0;
+            }
+
+            var result = new byte[decrypted.Length - pad];
+            Array.Copy(decrypted, 0, result, 0, decrypted.Length - pad);
+            return result;
+        }
+
+        /// <summary>
+        /// Decrypt the AES encrypted input string.
+        /// </summary>
+        /// <param name="input">Encrypted string.</param>
+        /// <param name="iv">Gets or sets the initialization vector for the symmetric algorithm.
+        //
+        // Returns:
+        //     The initialization vector.</param>
+        /// <param name="key"></param>
+        /// <returns></returns>
+        private byte[] AesDecrypt(string input, byte[] iv, byte[] key)
+        {
+            using (var aes = Aes.Create())
```

Comment:
Aes.Create() can somethings return null, validate it is not null and throw a CryptographicException if it is